pystac/extensions/base.py

```diff
@@ -190,7 +190,7 @@ def validate_owner_has_extension(
     @classmethod
     def ensure_owner_has_extension(
         cls,
-        asset: pystac.Asset | AssetDefinition,
+        asset_or_link: pystac.Asset | AssetDefinition | pystac.Link,
```

Comment:
This is a breaking change, isn't it? Since folks might be passing-by-keyword?

Ok if it is, we just would need to defer this until v2 then?

Reply:
I can leave it as being called `asset`. No strong sentiment on that.

Reply:
yeah it's note in v1.8.4, so maybe it's ok?